satpy/readers/ahi_hsd.py

```diff
@@ -316,45 +316,29 @@ def get_dataset(self, key, info):
 
     def get_area_def(self, dsid):
         del dsid
-        cfac = np.uint32(self.proj_info['CFAC'])
-        lfac = np.uint32(self.proj_info['LFAC'])
-        coff = np.float32(self.proj_info['COFF'])
-        loff = np.float32(self.proj_info['LOFF'])
-        a = float(self.proj_info['earth_equatorial_radius'] * 1000)
-        h = float(self.proj_info['distance_from_earth_center'] * 1000 - a)
-        b = float(self.proj_info['earth_polar_radius'] * 1000)
-        lon_0 = float(self.proj_info['sub_lon'])
-        nlines = int(self.data_info['number_of_lines'])
-        ncols = int(self.data_info['number_of_columns'])
-
-        # count starts at 1
-        cols = 1 - 0.5
-        left_x = (cols - coff) * (2.**16 / cfac)
-        cols += ncols
-        right_x = (cols - coff) * (2.**16 / cfac)
-
-        lines = (self.segment_number - 1) * nlines + 1 - 0.5
-        upper_y = -(lines - loff) * (2.**16 / lfac)
-        lines += nlines
-        lower_y = -(lines - loff) * (2.**16 / lfac)
-        area_extent = (np.deg2rad(left_x) * h, np.deg2rad(lower_y) * h,
-                       np.deg2rad(right_x) * h, np.deg2rad(upper_y) * h)
-
-        proj_dict = {'a': float(a),
-                     'b': float(b),
-                     'lon_0': float(lon_0),
-                     'h': float(h),
-                     'proj': 'geos',
-                     'units': 'm'}
-
-        area = geometry.AreaDefinition(
-            self.observation_area,
-            "AHI {} area".format(self.observation_area),
-            'geosh8',
-            proj_dict,
-            ncols,
-            nlines,
-            area_extent)
+
+        pdict = {}
+        pdict['cfac'] = np.uint32(self.proj_info['CFAC'])
+        pdict['lfac'] = np.uint32(self.proj_info['LFAC'])
+        pdict['coff'] = np.float32(self.proj_info['COFF'])
+        pdict['loff'] = np.float32(self.proj_info['LOFF'])
+        pdict['a'] = float(self.proj_info['earth_equatorial_radius'] * 1000)
+        pdict['h'] = float(self.proj_info['distance_from_earth_center'] * 1000 - pdict['a'])
+        pdict['b'] = float(self.proj_info['earth_polar_radius'] * 1000)
+        pdict['ssp_lon'] = float(self.proj_info['sub_lon'])
+        pdict['nlines'] = int(self.data_info['number_of_lines'])
+        pdict['ncols'] = int(self.data_info['number_of_columns'])
+        pdict['scandir'] = 'S2N'
```

Comment:
According to the video on this webpage, https://www.data.jma.go.jp/mscweb/en/himawari89/space_segment/spsg_ahi.html, AHI is scanning North to South. The fact that the computed area definition passes the AHI tests, I'd guess there is something wrong with the sign of the line offset.

Reply:
True. I have corrected this now.